unit_tests/testLinearSolver/testLinearSolverParallel.f90

```diff
@@ -0,0 +1,206 @@
+  CALL pList%add('LinearSolverType->x->VectorType->n',9_SNK)
+  CALL pList%add('LinearSolverType->b->VectorType->n',9_SNK)
+  CALL pList%validate(pList)
+  CALL thisLS%init(pList)
+
+  !A =  4    -1     0    -1     0     0     0     0     0
+  !    -1     4    -1     0    -1     0     0     0     0
+  !     0    -1     4     0     0    -1     0     0     0
+  !    -1     0     0     4    -1     0    -1     0     0
+  !     0    -1     0    -1     4    -1     0    -1     0
+  !     0     0    -1     0    -1     4     0     0    -1
+  !     0     0     0    -1     0     0     4    -1     0
+  !     0     0     0     0    -1     0    -1     4    -1
+  !     0     0     0     0     0    -1     0    -1     4
+  SELECTTYPE(A => thisLS%A); TYPE IS(DistributedBandedMatrixType)
+      CALL A%set(1,1, 4.0_SRK)
+      CALL A%set(1,2,-1.0_SRK)
+      CALL A%set(1,4,-1.0_SRK)
+      CALL A%set(2,1,-1.0_SRK)
+      CALL A%set(2,2, 4.0_SRK)
+      CALL A%set(2,3,-1.0_SRK)
+      CALL A%set(2,5,-1.0_SRK)
+      CALL A%set(3,2,-1.0_SRK)
+      CALL A%set(3,3, 4.0_SRK)
+      CALL A%set(3,6,-1.0_SRK)
+      CALL A%set(4,1,-1.0_SRK)
+      CALL A%set(4,4, 4.0_SRK)
+      CALL A%set(4,5,-1.0_SRK)
+      CALL A%set(4,7,-1.0_SRK)
+      CALL A%set(5,2,-1.0_SRK)
+      CALL A%set(5,4,-1.0_SRK)
+      CALL A%set(5,5, 4.0_SRK)
+      CALL A%set(5,6,-1.0_SRK)
+      CALL A%set(5,8,-1.0_SRK)
+      CALL A%set(6,3,-1.0_SRK)
+      CALL A%set(6,5,-1.0_SRK)
+      CALL A%set(6,6, 4.0_SRK)
+      CALL A%set(6,9,-1.0_SRK)
+      CALL A%set(7,4,-1.0_SRK)
+      CALL A%set(7,7, 4.0_SRK)
+      CALL A%set(7,8,-1.0_SRK)
+      CALL A%set(8,5,-1.0_SRK)
+      CALL A%set(8,7,-1.0_SRK)
+      CALL A%set(8,8, 4.0_SRK)
+      CALL A%set(8,9,-1.0_SRK)
+      CALL A%set(9,6,-1.0_SRK)
+      CALL A%set(9,8,-1.0_SRK)
+      CALL A%set(9,9, 4.0_SRK)
+      CALL A%assemble()
+  ENDSELECT
+
+  ! Use default initial guess (all ones)
+  SELECTTYPE(x => thisLS%x); TYPE IS(NativeDistributedVectorType)
+    x%b = 1.0_SRK
+  ENDSELECT
+
+  SELECTTYPE(b => thisLS%b); TYPE IS(NativeDistributedVectorType)
+    b%b = 1.0_SRK
+  ENDSELECT
+
+  !set iterations and convergence information and build/set M
+  SELECTTYPE(thisLS); TYPE IS(LinearSolverType_Iterative)
+    thisLS%hasX0 = .TRUE.
+    CALL thisLS%setConv(2_SIK,1.0E-6_SRK,1.0E-6_SRK,20_SIK,4_SIK)
+  ENDSELECT
+
+
+  !solve it
+  CALL thisLS%solve()
+
+  !Store expected solution (from MATLAB) in B
+  IF (mpiTestEnv%rank == 0) THEN
+    ALLOCATE(thisB(5))
+    thisB(1)=0.6875_SRK
+    thisB(2)=0.875_SRK
+    thisB(3)=0.6875_SRK
+    thisB(4)=0.875_SRK
+    thisB(5)=1.125_SRK
+    thisB=10000.0_SRK*thisB
+    match=.TRUE.
+    DO i=1,SIZE(thisB)
+      SELECTTYPE(X => thisLS%X); TYPE IS(NativeDistributedVectorType)
+        IF(ALLOCATED(dummyvec)) DEALLOCATE(dummyvec)
+        ALLOCATE(dummyvec(SIZE(X%b)))
+        dummyvec = X%b
+        IF(NINT(thisB(i)) /= NINT(10000.0_SRK*dummyvec(i))) THEN
```

Comment:
This may be obvious to someone, not me, why multiply both test and reference values by 10,000 and `nint`? If it is just about tolerance why not `softeq`?

Reply:
It should absolutely be softeq. Some of these edits are quite old, when I was first learning the code base. In this case, I just copied existing tests but I will go back and update them all to use the new (since the original files were written anyways) methods. This NINT method is a bit archaic.

Reply:
did this get updated?

Reply:
Yes, the unit test now uses ALL's and APPROXEQ's so these checks are much cleaner.